### PR TITLE
lines_cop.rb: Disable OS.mac? audit for Linuxbrew

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
Since brew 1.3.4, each time OS.mac? is used, we get the following audit error:
```
Don't use OS.mac?; Homebrew/core only supports macOS
```

This new test only makes sense on mac OS, so we can disable it for Linuxbrew.

**Comment by iMichka** [comment]
```
Failures:

  1) RuboCop::Cop::FormulaAudit::Miscellaneous When auditing formula with OS.linux? check
     Failure/Error: expect_offense(expected, actual)

       expected: not nil
            got: nil
     # ./test/support/helper/rubocop.rb:5:in `expect_offense'
     # ./test/rubocops/lines_cop_spec.rb:252:in `block (4 levels) in <top (required)>'
     # ./test/rubocops/lines_cop_spec.rb:251:in `each'
     # ./test/rubocops/lines_cop_spec.rb:251:in `block (3 levels) in <top (required)>'
     # ./test/spec_helper.rb:96:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'
```

**Review by sjackman — APPROVED** [review]

**Comment by sjackman** [comment]
```
Failures:

  1) Bash every `.sh` file has valid bash syntax
     Failure/Error: expect(path).to have_valid_bash_syntax

       expected that /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/shcmd.sh is a valid Bash file:
       /bin/bash: /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/shcmd.sh: No such file or directory
     # ./test/bash_spec.rb:29:in `block (4 levels) in <top (required)>'
     # ./test/bash_spec.rb:25:in `each'
     # ./test/bash_spec.rb:25:in `block (3 levels) in <top (required)>'
     # ./test/spec_helper.rb:96:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'
```
I believe this failure is intermittent. It depends on the order in which the tests are run. I'm rerunning CircleCI now.

File: Library/Homebrew/rubocops/lines_cop.rb [hunk 1]
```diff
@@ -79,7 +79,7 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
             find_instance_method_call(body_node, "OS", method_name) do |check|
               problem "Don't use #{check.source}; Homebrew/core only supports macOS"
             end
-          end
+          end if OS.mac?
```
Comment (sjackman) [inline Library/Homebrew/rubocops/lines_cop.rb:82]: Please add `next unless OS.mac?` after line 77, before the existing `next unless`.

File: Library/Homebrew/test/rubocops/lines_cop_spec.rb [hunk 2]
```diff
@@ -251,7 +251,7 @@ class Foo < Formula
       expected_offenses.zip(cop.offenses).each do |expected, actual|
         expect_offense(expected, actual)
       end
-    end
+    end if OS.mac?
```
Comment (sjackman) [inline Library/Homebrew/test/rubocops/lines_cop_spec.rb:254]: Try changing line 229 to
```ruby
it "with OS.linux? check", :needs_macos do
```
I'm not certain that's the correct syntax, but give it a go.

**Review by sjackman — CHANGES_REQUESTED** [review]

**Comment by sjackman** [comment]
CircleCI succeeded this time, so it is an intermittent failure.

File: Library/Homebrew/rubocops/lines_cop.rb (hunk 1 above)
Comment (sjackman) [reply Library/Homebrew/rubocops/lines_cop.rb:82]: I turns out that `OS.mac?` is not available inside this function, that is, when running Rubocop. This file is a plugin to Rubocop, and is not run inside the usual Brew environment.

**Comment by sjackman** [comment]
Supersed by https://github.com/Linuxbrew/brew/pull/503

**Comment by iMichka** [comment]
Thanks @sjackman

**Pull request closed without merging**